observer/probers/crl/crl.go

```diff
@@ -0,0 +1,56 @@
+package probers
+
+import (
+	"io/ioutil"
+	"net/http"
+	"time"
+
+	"github.com/letsencrypt/boulder/crl/crl_x509"
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+// CRLProbe is the exported 'Prober' object for monitors configured to
+// monitor CRL availability & characteristics.
+type CRLProbe struct {
+	url         string
+	cNextUpdate *prometheus.GaugeVec
+	cThisUpdate *prometheus.GaugeVec
+	cCertCount  *prometheus.GaugeVec
+}
+
+// Name returns a string that uniquely identifies the monitor.
+func (p CRLProbe) Name() string {
+	return p.url
+}
+
+// Kind returns a name that uniquely identifies the `Kind` of `Prober`.
+func (p CRLProbe) Kind() string {
+	return "CRL"
+}
+
+// Probe requests the configured CRL and publishes metrics about it if found.
+func (p CRLProbe) Probe(timeout time.Duration) (bool, time.Duration) {
+	start := time.Now()
+	resp, err := http.Get(p.url)
+	if err != nil {
+		return false, time.Since(start)
+	}
+
+	body, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return false, time.Since(start)
+	}
+
+	crl, err := crl_x509.ParseRevocationList(body)
+	if err != nil {
+		return false, time.Since(start)
+	}
+
+	dur := time.Since(start)
+
+	p.cThisUpdate.WithLabelValues(p.url).Set(float64(crl.ThisUpdate.Unix()))
+	p.cNextUpdate.WithLabelValues(p.url).Set(float64(crl.NextUpdate.Unix()))
```

Comment:
We should use the 64 bit timestamp here or come 2038 we'll be writing metrics like it's 1901:
```suggestion
	p.cThisUpdate.WithLabelValues(p.url).Set(float64(crl.ThisUpdate.UnixNano()))
	p.cNextUpdate.WithLabelValues(p.url).Set(float64(crl.NextUpdate.UnixNano()))
```

Reply:
I believe the difference between `UnixNano` and `Unix` is that the latter returns seconds and the former returns nanoseconds. According to these [docs](https://pkg.go.dev/time#Time.Unix), I think they both return a 64-bit timestamp. Here's the doc for `Unix`:

> Unix returns t as a Unix time, the number of seconds elapsed since January 1, 1970 UTC. The result does not depend on the location associated with t. Unix-like operating systems often record time as a 32-bit count of seconds, but since the method here returns a 64-bit value it is valid for billions of years into the past or future.

In which case, seconds are likely more than sufficient for any monitoring/alerting use-case.

Reply:
I actually missed that, my eyes got scanned the '32-bit' in the description my mind was made up.

Additionally, I've checked the docs and that Prometheus `time()` func expects seconds, so this is good. Unrelated to your implementation, if you plan on parsing a Unix timestamp (from Prometheus) as a datetime in a dashboard you'll have to multiply it by `1000` as Grafana expects nanoseconds.